sn_networking/src/replication_fetcher.rs

```diff
@@ -39,6 +39,8 @@ pub(crate) struct ReplicationFetcher {
     // Avoid fetching same chunk from different nodes AND carry out too many parallel tasks.
     on_going_fetches: HashMap<(RecordKey, RecordType), (PeerId, ReplicationTimeout)>,
     event_sender: mpsc::Sender<NetworkEvent>,
+    // Distance range that the incoming key shall be fetched
```

Comment:
Is this more like the `farthest record we hold`?

Reply:
not exactly.
for incoming new data, we store it anyway (to give margin for the client payee selection and allow the record to be `replicated out`, if the picked payee is an `edging` one)
hence, there is chance that `the incoming record is farther than the distance_range, but we still have to store it`, even though that record will have a high chance got `pruned` within the next round of pruning.